### PR TITLE
Start using code-formatters

### PR DESCRIPTION
**Pull request by WilliamJamieson**
This PR applies some code formatters to `roman_datamodels`. Namely:

- [`isort`](https://pycqa.github.io/isort/) to sort the imports.
- [`black`](https://black.readthedocs.io/en/stable/) to consistently format the code.

Note that this PR is based off #119, which should be merged first.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -93,3 +93,21 @@ exclude_lines = [
     'def main\(.*\):',
     'if __name__ == \(.*\):',
 ]
+
+[tool.isort]
+profile = "black"
+filter_files = true
+line_length = 130
```
Comment (WilliamJamieson) [inline pyproject.toml:100]:
```suggestion
line_length = 130
```

I am not a fan of such a long `line_length`; however, this was the default configured by `flake8` so I stuck with it.

I would rather see something akin to the `black` default of `88` characters or the default for `flake8` of `80`. The long line length makes it annoying when viewing the code in a narrow window.

File: docs/conf.py [hunk 2]
```diff
@@ -33,98 +32,95 @@ def setup(app):
...
 
 def check_sphinx_version(expected_version):
     sphinx_version = LooseVersion(sphinx.__version__)
     expected_version = LooseVersion(expected_version)
     if sphinx_version < expected_version:
         raise RuntimeError(
-            "At least Sphinx version {0} is required to build this "
-            "documentation.  Found {1}.".format(
-                expected_version, sphinx_version))
+            f"At least Sphinx version {expected_version} is required to build this " f"documentation.  Found {sphinx_version}."
```
Comment (PaulHuwe) [inline docs/conf.py:54]: Why the multiple f-strings on the same line?

File: src/roman_datamodels/datamodels.py [hunk 3]
```diff
@@ -62,39 +62,35 @@ def __init__(self, init=None, **kwargs):
...
+            self._instance = asdffile.tree["roman"]
         elif isinstance(init, stnode.TaggedObjectNode):
             self._instance = init
             asdffile = asdf.AsdfFile()
-            asdffile.tree = {'roman': init}
+            asdffile.tree = {"roman": init}
         else:
-            raise IOError("Argument does not appear to be an ASDF file"
-                          " or TaggedObjectNode.")
+            raise OSError("Argument does not appear to be an ASDF file" " or TaggedObjectNode.")
```
Comment (PaulHuwe) [inline src/roman_datamodels/datamodels.py:76]: Another split string.

File: src/roman_datamodels/util.py [hunk 4]
```diff
@@ -2,27 +2,28 @@
...
 import traceback
+from platform import system as platform_system
 from pydoc import locate
 
-#from . import s3_utils
-#from .basic_utils import bytes2human
+import psutil
+
+# from . import s3_utils
+# from .basic_utils import bytes2human
```
Comment (PaulHuwe) [inline src/roman_datamodels/util.py:14]: Should these lines be deleted?

File: src/roman_datamodels/util.py [hunk 5]
```diff
@@ -562,11 +552,9 @@ def get_available_memory_darwin(include_swap=True):
 
         # Attempt to determine amount of free disk space on the boot partition.
         try:
-            swap = psutil.disk_usage('/private/var/vm').free
+            swap = psutil.disk_usage("/private/var/vm").free
         except FileNotFoundError as exception:
-            log.warn('Cannot determine available swap space.'
-                     f'Reason:\n'
-                     f'{"".join(traceback.format_exception(exception))}')
+            log.warn("Cannot determine available swap space." f"Reason:\n" f'{"".join(traceback.format_exception(exception))}')
```
Comment (PaulHuwe) [inline src/roman_datamodels/util.py:557]: Multiple f-strings on one line.

File: tests/test_open.py [hunk 6]
```diff
@@ -116,22 +123,35 @@ def test_memmap(tmp_path):
...
+        np.zeros(
+            (
+                400,
+                400,
+            ),
+            dtype=np.float32,
+        ),
+        ru.electron / u.s,
+        dtype=np.float32,
+    )
```
Comment (PaulHuwe) [inline tests/test_open.py:145]: Why is this fully expanded and other groupings are completely compressed? (e.g. test_stdone new lines 76-80)

**Review by PaulHuwe — CHANGES_REQUESTED** [review]

File: src/roman_datamodels/util.py (hunk 4 above)
Comment (WilliamJamieson) [reply src/roman_datamodels/util.py:14]: Sure, I can do that. There are some more extreme `ruff` checks that will autofix and/or flag this stuff.

**Review by PaulHuwe — APPROVED** [review]
LGTM

**Pull request merged**